### PR TITLE
Replace DataOrErrors with Result

### PR DESCRIPTION
**Pull request by tjmaynes**
## Motivation:

Replacing `DataOrErrors` will enable developers to utilize a library (`result-ts`) that manages to accomplish everything that `DataOrErrors` strives to achieve without having to maintain extra code (`DataOrErrors`).

### CODE REVIEW

File: src/ApiRequestComponent.tsx [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import React, { ReactElement, useEffect, useState } from 'react'
 import MainWrapper from './MainWrapper'
-import { DataOrErrors } from './api_clients/FetchTypes'
 import { Card } from 'react-bootstrap'
 import Loadable, { LoadingState } from './Loadable'
+import {Result} from "ts-results";
+import {ResponseError} from "./api_clients/FetchBroker";
```
Comment (mkollasch-vmware) [inline src/ApiRequestComponent.tsx:6]: Please run the autoformatter again. I'm surprised the linter didn't complain about the spacing within the braces here.

File: src/Loadable.tsx [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
-}: LoadableProps<T>): ReactJSXElement {
+}: LoadableProps<T, E>): ReactJSXElement {
   if (!state) return errorComponent
   else if ('isLoading' in state) return loadingComponent
-  else if ('errors' in state)
-    return React.cloneElement(errorComponent, { errors: state.errors })
-  else if (children) return React.cloneElement(children, { data: state.data })
+  else if (state.err)
+    return React.cloneElement(errorComponent, { errors: state.val })
+  else if (children) return React.cloneElement(children, { data: state.val })
```
Comment (mkollasch-vmware) [inline src/Loadable.tsx:28]: Is this transforming the `Result` into a `DataOrErrors`? If `Result` has e.g. a `map` and a `mapErr` method, those might more clearly express the desired transformation here.

File: src/api_clients/AnalyticsClient.test.ts [hunk 3]
```diff
@@ -4,49 +4,49 @@ import {
...
 } from '../test_utils/IntegrationTestApiServerTools'
+import {Err, Ok} from "ts-results";
+
 const server = setupIntegrationTestApiServer()
 
 describe('Analytics Client', function () {
   it('should call .json before then getting the data out of the data key before returning', async function () {
     const goodAnalyticsResponse = { data: { key: 'any string we want here' } }
-    const expected = { data: goodAnalyticsResponse.data.key }
+    const expected = goodAnalyticsResponse.data.key
```
Comment (mkollasch-vmware) [inline src/api_clients/AnalyticsClient.test.ts:14]: I suggest setting `expected = Ok(goodAnalyticsResponse.data.key)` in order to preserve the structure of the test, and likewise below.

**Review by mkollasch-vmware — CHANGES_REQUESTED** [review]
Overall good. Just needs the autoformatter and one possible case of missed conversion. I'll take care of it.

File: src/Loadable.tsx (hunk 2 above)
Comment (tjmaynes) [reply src/Loadable.tsx]: Good call, I've updated the code to utilize `ts-results` functional approach:

```ts
state
    .map((result) => React.cloneElement(children, { data: result }))
    .mapErr((err) => React.cloneElement(errorComponent, { errors: err }))
    .unwrapOr(errorComponent)
```

Comment (mkollasch-vmware) [reply src/Loadable.tsx]: Upon closer inspection, this is expressing an expectation that the child component's props extend `{ data: T }`. Annotating the props of `Loadable` and `ApiRequestComponent` accordingly would more clearly express this intention.

**Review by mkollasch-vmware — APPROVED** [review]

**Pull request merged**